### PR TITLE
What's new in 2.4.0

### PR DESCRIPTION
**Pull request by micheles**
See https://github.com/gem/oq-engine/blob/whatsnew/doc/whats-new.md

### CODE REVIEW

File: doc/dropping-python-2.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+Python 3 came out nearly 10 years ago, Python 2.7 is not really supported
+anymore - no bug fixes are accepted except security bugs - and it will reach
+the end of line in 2020. For these reasons [several projects have
+already dropped support for it](http://www.python3statement.org/).
+In particular, the following projects that we rely upon have already dropped
+support for Python 2.7 or they will do it shortly:
+
+- ipython
+- jypiter
```
Comment (ptormene) [inline doc/dropping-python-2.md]: jupyter

File: doc/whats-new.md [hunk 2]
```diff
@@ -1,200 +1,476 @@
...
+  output generated when you set `asset_loss_table=true`.
+- There is a new output `losses by event` produced by the `scenario_risk`
+  calculator, which is similar to the event loss table in the event based
+  calculator: it contains the aggregate losses for each seismic event.
+  There is a new CSV exporter for it.
+- There are new outputs for the losses aggregated by taxonomy which are
+  produced by the calculators `scenario_risk`, `event_based_risk` and
+  `ucerf_risk`. There are CSV exporters associated.
+- There is a new CSV exporter for the loss curves, while the old one has
+  been removed
```
Comment (ptormene) [inline doc/whats-new.md]: Missing `.`

File: doc/whats-new.md [hunk 3]
```diff
@@ -1,200 +1,476 @@
...
+a 64 bit array: this reduces the memory consumption and data storage
+by half. Moreover, we have now an efficient way to serialize ruptures
+into HDF5 format.
+
+We fixed a numerical issue involving the square root of small negative numbers,
+due to rounding issues.
+
+We have refactored the filtering mechanism and we have now a
+single `IntegrationDistance` class in charge of filtering out sites and
+ruptures outside of the integration distance. For safe of correctness,
```
Comment (ptormene) [inline doc/whats-new.md]: safe -> sake (I suppose)

File: doc/whats-new.md [hunk 4]
```diff
@@ -1,200 +1,476 @@
...
-users with a cluster.
+We added an endpoint `GET /v1/available_gsims` to the REST API underlying
+the WebUI. This is used by the OpenQuake platform to extract the list
+of available GSIM classes. Now the platform uses the engine as a service
+and it does not import directly any code from it.
 
-Changes
-------------
+We changed the Web UI button from "Run Risk" to "Continue", since it
+can also be used or postprocessing of hazard calculations.
```
Comment (ptormene) [inline doc/whats-new.md]: `or` -> `for`, and it can also be used for postprocessing of risk calculations

File: doc/dropping-python-2.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+We could drop support for Python 2 even in the next release, but in
+consideration for our users we will keep supporting it until the end
+of 2017. In the course of 2018, however, we will start using Python 3
+features. This means that users of hazardlib and the engine will have
+to upgrade to Python 3 to stay updated. This requirement only applies to
+users *importing* directly Python modules. If you are using the Web
+API of the engine, you will be able to keep your client in Python 2,
+even if the server will require Python 3. Still, it is a good idea to
+migrate. Python 2.7 is dying.
```
Comment (daniviga) [inline doc/dropping-python-2.md]: Maybe we can add a link to our issue about Py 2 -> Py 3 packages migration. Users and even more sysadmins need a medium term road map for such big changes.

Comment (daniviga) [inline doc/dropping-python-2.md]: https://github.com/gem/oq-engine/issues/2803

File: doc/whats-new.md [hunk 6]
```diff
@@ -1,200 +1,476 @@
...
+Make init scripts more robust and portable
+Turned matplolib into a required library and added a plotting test on Travis
+Check if oq is talking to a foreign DbServer
+Sync packager.sh with oq-hazardlib to manage oq-libs-extra
+Bump to h5py 2.7.0
+Differentiate listening interface and connect interface in dbserver
+Small fixes for local_settings template to be able to append settings
+Add a template for PAM auth
+Split RPM packages in standard, master, worker
+Use CDN ftp mirror instead of upstream
```
Comment (daniviga) [inline doc/whats-new.md]: I would keep only:

Change how `local_settings.py` is found
I need to explain the logic

Turned matplolib into a required library and added to our distributions

 Check if oq is talking to a foreign DbServer

Bump to h5py 2.7.0

Add a template for PAM auth

Split RPM packages in standard, master, worker

Everything else isn't for users and may be only bring more confusion

File: doc/dropping-python-2.md (hunk 1 above)
Comment (micheles) [inline doc/dropping-python-2.md]: Fixed

File: doc/whats-new.md (hunk 2 above)
Comment (micheles) [inline doc/whats-new.md]: Fixed

File: doc/whats-new.md (hunk 3 above)
Comment (micheles) [inline doc/whats-new.md]: Fixed

File: doc/whats-new.md (hunk 4 above)
Comment (micheles) [inline doc/whats-new.md]: Fixed

File: doc/dropping-python-2.md (hunk 5 above)
Comment (micheles) [inline doc/dropping-python-2.md]: Done

**Pull request merged**